presto-hive-functions/src/test/java/com/facebook/presto/hive/functions/TestScalarMethodHandles.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.functions;
+
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.common.type.TestingTypeManager;
+import com.facebook.presto.common.type.TypeSignature;
+import com.facebook.presto.hive.functions.scalar.ScalarFunctionInvoker;
+import com.facebook.presto.spi.function.FunctionKind;
+import com.facebook.presto.spi.function.Signature;
+import io.airlift.slice.Slices;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandle;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.hive.functions.gen.ScalarMethodHandles.generateUnbound;
+import static org.testng.Assert.assertEquals;
+
+public class TestScalarMethodHandles
+{
+    private final AtomicInteger number = new AtomicInteger(0);
+    private final TestingTypeManager typeManager = new TestingTypeManager();
+
+    @Test
+    public void generateUnboundTest()
+            throws Throwable
+    {
+        runCase("boolean", tokens("boolean"), true, false);
```

Comment:
can we make sure we cover all supported hive types, including array / map?

Reply:
oh. It has only been designed for testing the generateUnbound logic. I put type-convertion test in `TestObjectEncoders` and `TestObjectInputDecoders`. But for the integrity, I add the array and map case here too.